RadixWallet/Features/TransactionReviewFeature/TransactionReview+Sections.swift

```diff
@@ -679,20 +826,34 @@ extension TransactionReview {
 	private func stakeClaimTransfer(
 		_ resource: OnLedgerEntity.Resource,
 		stakeClaimValidator: OnLedgerEntity.Validator,
+		unstakeData: [UnstakeDataEntry],
 		tokens: [OnLedgerEntity.NonFungibleToken]
 	) throws -> [Transfer] {
-		struct InvalidStakeClaimToken: Error {}
+		let stakeClaimTokens: [OnLedgerEntitiesClient.StakeClaim] = if !unstakeData.isEmpty {
+			try tokens.map { token in
+				guard let data = unstakeData.first(where: { $0.nonFungibleGlobalId == token.id })?.data else {
+					throw MissingStakeClaimTokenData()
+				}
 
-		let stakeClaimTokens = try tokens.map { token in
-			guard let data = token.data, let claimAmount = data.claimAmount else {
-				throw InvalidStakeClaimToken()
+				return OnLedgerEntitiesClient.StakeClaim(
+					validatorAddress: stakeClaimValidator.address,
+					token: token,
+					claimAmount: data.claimAmount,
+					reamainingEpochsUntilClaim: Int(data.claimEpoch) - Int(resource.atLedgerState.epoch)
```

Comment:
for the case of stake claim resource as part of unstake transaction, we can/should set `reamainingEpochsUntilClaim: nil`.